test/python/test_select.py

```diff
@@ -64,3 +64,44 @@ def run_test(sess):
 
         assert (
             self.with_ngraph(run_test) == self.without_ngraph(run_test)).all()
+
+    def test_select_complexshape1(self):
+        a = np.random.randint(2, size=[7])
+        x = np.random.uniform(0, 11, [7, 3, 2, 1])
+
+        p = tf.placeholder(dtype=tf.bool)
+        out = tf.where(p, x, x)
+
+        def run_test(sess):
+            return (sess.run(out, feed_dict={p: a})).shape
+
+        assert (self.with_ngraph(run_test) == self.without_ngraph(run_test))
+
+    def test_select_complexshape2(self):
+        a = np.random.randint(2, size=[5])
+        x = np.random.uniform(0, 11, [5, 3, 1])
+
+        p = tf.placeholder(dtype=tf.bool)
+        out = tf.where(p, x, x)
+
+        def run_test(sess):
+            return (sess.run(out, feed_dict={p: a})).shape
+
+        assert (self.with_ngraph(run_test) == self.without_ngraph(run_test))
+
+    #failin test case
+    @pytest.mark.skip(
+        reason=
+        "First dimension of input tensor should match the shape of condition tensor"
+    )
+    def test_select_fail(self):
+        a = np.random.randint(2, size=[5])
+        x = np.random.uniform(0, 11, [4, 3, 1])
+
+        p = tf.placeholder(dtype=tf.bool)
+        out = tf.where(p, x, x)
+
+        def run_test(sess):
+            return (sess.run(out, feed_dict={p: a})).shape
```

Comment:
why are we returning `shape` from `run_test`. Shouldn't we return the actual data tensor?

Reply:
ah! i did this for debugging purpose to print shapes. fixed this!